pkg/controllers/multiclusterservice/mcs_controller.go

```diff
@@ -119,6 +119,46 @@ func (c *MCSController) Reconcile(ctx context.Context, req controllerruntime.Req
 	return controllerruntime.Result{}, nil
 }
 
+func (c *MCSController) checkClusterConfiguration(mcs *networkingv1alpha1.MultiClusterService,
+	providerClusters, noneExistProviderClusters, consumerClusters, noneExistConsumerClusters sets.Set[string]) error {
+	if len(noneExistProviderClusters) != 0 || len(noneExistConsumerClusters) != 0 {
+		msgProvider := strings.Join(noneExistProviderClusters.UnsortedList(), ",")
+		msgConsumer := strings.Join(noneExistConsumerClusters.UnsortedList(), ",")
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeNormal, events.EventReasonConfigurationRedundant,
+			fmt.Sprintf("ProviderClusters(%s)/ConsumerClusters(%s) dont's exist, Karmada will ignore them", msgProvider, msgConsumer))
+	}
+
+	if !helper.MultiClusterServiceCrossClusterEnabled(mcs) {
+		return nil
+	}
+
+	clusters := clusterv1alpha1.ClusterList{}
+	if err := c.Client.List(context.Background(), &clusters); err != nil {
+		klog.Errorf("Failed to list clusters:%v", err)
+		return err
+	}
+
+	clusterDisabledEndpointSlice := []string{}
+	for _, cluster := range clusters.Items {
+		if !providerClusters.Has(cluster.Name) && !consumerClusters.Has(cluster.Name) {
+			continue
+		}
+
+		endpointSliceGVK := discoveryv1.SchemeGroupVersion.WithKind("EndpointSlice")
+		if !helper.IsAPIEnabled(cluster.Status.APIEnablements, endpointSliceGVK.GroupVersion().String(), endpointSliceGVK.Kind) {
+			clusterDisabledEndpointSlice = append(clusterDisabledEndpointSlice, cluster.Name)
+		}
+	}
+
+	if len(clusterDisabledEndpointSlice) != 0 {
+		msg := strings.Join(clusterDisabledEndpointSlice, ",")
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventReasonConfigurationRedundant,
```

Comment:
Here using `EventReasonConfigurationRedundant` may not be reasonable.

Reply:
How about EventReasonAPIIncompatible